packages/edit-post/src/components/header/header-toolbar/index.native.js

```diff
@@ -185,6 +185,12 @@ export default compose( [
 			onHideKeyboard() {
 				clearSelectedBlock();
 				togglePostTitleSelection( false );
+				// Explicitly dismiss the keyboard for circumstances where Aztec race
+				// conditions lead to one block's rich text focused, but a different
+				// non-rich-text block selected. In this context, merely clearing block
+				// selection does not trigger a blur event and therefore will not
+				// dismiss the keyboard.
+				Keyboard.dismiss();
```

Comment:
This change reintroduces an explicit keyboard dismissal that was removed in https://github.com/WordPress/gutenberg/commit/54323e94b843ad28e720ac1a095c8a981d287908 as it was deemed unnecessary. It _should_ be unnecessary, but there are race conditions between Aztec and React Native event handlers (e.g. block/rich text tap) that makes it possible to focus one block and select another when tapping the two near simultaneously. 

Explicitly dismissing the keyboard ensures that the user is able to close the keyboard when they find themselves in a state where a non-rich-text block is selected with the keyboard visible (the second image below).

| Rich Text | Non Rich Text |
| - | - |
| ![differing-focus-and-rich-text-block-select](https://user-images.githubusercontent.com/438664/199586443-64e96ee2-a85d-4f78-a4ef-7df78ca46ebe.jpeg) | ![differing-focus-and-non-rich-text-block-select](https://user-images.githubusercontent.com/438664/199586472-79d2d334-1cbe-4d02-91af-5dbe2b61368f.jpeg) |

This issue is not new in this PR, it is just more accessible now that focus loops are avoided on iOS. The change on this line fixes the focus + selection conflict on iOS. Unfortunately, it does not fix the issue on Android for an unknown reason currently.